src/lib.rs

```diff
@@ -220,13 +250,35 @@ impl From<Type> for Value {
     }
 }
 
+impl Type {
+    pub fn fits(&self, other: &Self) -> bool {
+        match (self, other) {
```

Comment:
You could add a comment explaining what `fits` does since the name is a bit weird.